### DISCUSSION TITLE
Why doesn't graphics.transform update the matrix of its internal GraphicsPath

### DISCUSSION THREAD

**Discussion by oubenruing** [pixijs/pixijs#11091] [General]
I previously mentioned an [issue](https://github.com/pixijs/pixijs/issues/11026) that I initially thought was a bug. After reading the code, I found that Graphics.transform doesn’t update the matrix of its internal GraphicsPath. I’d like to understand the reasoning behind this design.

Additionally, if I want to update the matrix after Graphics executes a command, what would be the best practice?

Here’s a simple use case: I draw a line and then interact with it via mouse dragging to change its size or position. In this case, I want to directly pass a matrix to modify the line’s vertex coordinates. 
```
    const graphics = new Graphics();
    graphics.moveTo(10, 10);
    graphics.lineTo(20, 20);
    app.stage.addChild(graphics)
    // mouseEvent ····
    graphics.transform(new Matrix().scale(5,1))
```

Currently, it seems like I have to re-new a Graphics. Would this approach affect performance?

```
    const graphics = new Graphics();
    graphics.moveTo(10, 10);
    graphics.lineTo(20, 20);
    app.stage.addChild(graphics)

   // mouseEvent ····
   const graphics = new Graphics();
   graphics.transform(new Matrix().scale(5,1))
   graphics.moveTo(10, 10);
   graphics.lineTo(20, 20);
   app.stage.addChild(graphics)
    
```

Thanks.

**Comment by GoodBoyDigital** [comment]
hey there! I think [this guide](https://pixijs.com/8.x/guides/components/graphics) will help you to understand how the graphics API is intended to work. Let me know if you have any further questions after reading!

**Comment by oubenruing** [comment]
Hi! I've read the documentation and looked at some of the code. I believe I now understand how Graphics works: when executing methods like MoveTo and lineTo, Graphics essentially creates drawing commands. My question is, graphics.transform() only affects commands created after it is called. If I want to modify the commands created before calling graphics.transform(), how should I go about doing that?

The reason I need this feature instead of directly using methods like scale is that I want the stroke width of the shapes I create to remain unaffected by the scale.Unlike pixelLine: true, I still need the graphic to respond to the scaling of its parent element.


In fact, I noticed that the [GraphicsPath](https://github.com/pixijs/pixijs/blob/6c08915bf8e585260852e92762c13fa557fd97bf/src/scene/graphics/shared/path/GraphicsPath.ts#L627) has a transform method that can update the coordinates of already generated commands, but this method is not exposed through the Graphics object. I'm not sure if calling it directly would work.
```
    graphics.moveTo(10, 10);
    graphics.lineTo(100, 10);
    // There might be more points...
    graphics.lineTo(100, 20);
    graphics.stroke({ width: 4, color: 0xffd900 });
    
    // Is there any way to modify the commands generated by the code above? 
    // Maybe something like this
    // graphics._context._activePath.transform(matrix)
```

Thanks.

**Reply by bigtimebuddy** [reply]
The best way to achieve this is to clear and redraw the graphics. Unlike SVG, a stroke in Pixi is interpreted as triangles. The only way to change the width of a path depending on the parent scale is to re-triangulate the path.

**Reply by oubenruing** [reply]
Thanks for your reply! I’ve been using Graphics for a while now, and I think I now understand the design approach behind it. 😄

**Reply by bigtimebuddy** [reply]
This is a pretty common question for folks wanting to dynamically change Graphics' attributes (color, stoke, etc), and it's not designed to be manipulated after drawing other than the Container's transform, alpha or tint. Graphics has different constraints than SVG when rendering. Clearing is fine, but can get performantly expensive if doing it frequently (every frame). Hope that helps!
